format/leveldb/leveldb_table.go

```diff
@@ -82,9 +82,13 @@ func ldbTableDecode(d *decode.D) any {
 	var metaIndexOffset int64
 	var metaIndexSize int64
 
-	d.SeekAbs(d.Len() - footerEncodedLength)
 	d.FieldStruct("footer", func(d *decode.D) {
-		handleLength := d.LimitedFn(footerEncodedLength, func(d *decode.D) {
+		// check for magic number and fail fast if it isn't there
+		d.SeekAbs(d.Len() - magicNumberLength)
+		d.FieldU64("magic_number", d.UintAssert(tableMagicNumber), scalar.UintHex)
```

Comment:
If you want you can probably keep it like it was, as long as it does not decode megabytes of data before failing it should probably be ok.